### PR TITLE
Enable test_wpt.py to be run with python3

### PR DESCRIPTION
**Pull request by svillar**
The main issue while making it py2/py3 compatible was the writing of the manifest. All the code was based on handing bytes. However the json module does only generate str output not bytes. That was fine for python2 (bytes==str) but invalid for python3.

Most of the work was related to switch code to use str instead of bytes as required. The manifest file is also no open in binary mode for writting since we are not longer providing bytes.

### CODE REVIEW

**Comment by svillar** [comment]
Fixing the linting issues...

**Comment by stephenmcgruer** [comment]
Re-assigning since I'm OOO for a bit. But looks like there are both issues as well as a bunch of test failures? (e.g. https://community-tc.services.mozilla.com/tasks/Wam-iboKTCq310yTM0OtXg/runs/0/logs/https%3A%2F%2Fcommunity-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FWam-iboKTCq310yTM0OtXg%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log). Might just be the lint issues, haven't looked closely.

File: tools/manifest/sourcefile.py [hunk 1]
```diff
@@ -89,8 +89,8 @@ def get_any_variants(item):
     """
     Returns a set of variants (bytestrings) defined by the given keyword.
```
Comment (jgraham) [inline tools/manifest/sourcefile.py]: This comment and one below are now wrong.

File: tools/manifest/sourcefile.py [hunk 2]
```diff
@@ -209,7 +209,7 @@ def __init__(self, tests_root, rel_path, url_base, hash=None, contents=None):
         if os.name == "nt":
             # do slash normalization on Windows
             if isinstance(rel_path, binary_type):
-                rel_path = rel_path.replace(b"/", b"\\")
+                rel_path = rel_path.replace("/", "\\")
```
Comment (jgraham) [inline tools/manifest/sourcefile.py]: These two codepaths now look to do the same thing on Python 3. Is one actually unused?

File: tools/manifest/sourcefile.py [hunk 3]
```diff
@@ -233,7 +233,7 @@ def __init__(self, tests_root, rel_path, url_base, hash=None, contents=None):
         self.url_base = url_base
         self.contents = contents
         self.items_cache = None  # type: Optional[Tuple[Text, List[ManifestItem]]]
-        self._hash = hash
+        self._hash = ensure_str(hash, encoding='ascii') if hash else None
```
Comment (jgraham) [inline tools/manifest/sourcefile.py]: The mypy annotation makes it look like this should always be Text on input. Why do we end up converting to bytes in Python 2?

**Review by jgraham — CHANGES_REQUESTED** [review]
I'm worried by the use of `ensure_str` everywhere and the `str` type. I'd really prefer that we have some invariant about the data either being Text or Binary and uphold that in both Python 2 and Python 3, rather than having different types in the two cases. I also think we should document what the invariants/strategy is in the commit message an in a comment.

File: tools/manifest/sourcefile.py (hunk 2 above)
Comment (svillar) [reply tools/manifest/sourcefile.py]: I guess you're right

File: tools/manifest/sourcefile.py (hunk 3 above)
Comment (svillar) [reply tools/manifest/sourcefile.py]: Indeed, seems a leftover of one the various approaches I took

**Comment by svillar** [comment]
> I'm worried by the use of `ensure_str` everywhere and the `str` type. I'd really prefer that we have some invariant about the data either being Text or Binary and uphold that in both Python 2 and Python 3, rather than having different types in the two cases. I also think we should document what the invariants/strategy is in the commit message an in a comment.

yeah, I am not happy either with the current approach. I'll try to limit the changes just to the JSON export phase and leave the core untouched if possible

**Comment by svillar** [comment]
Closing to try another approach

**Pull request closed without merging**